druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java

```diff
@@ -122,6 +125,10 @@ public String apply(@NotNull String key, @NotNull String valueColumn, DecodeConf
                     LOG.error("RocksDB instance is null");
                     LOG.error("Failed to get lookup value from cache. Falling back to lookupService.");
                     serviceEmitter.emit(ServiceMetricEvent.builder().build(MonitoringConstants.MAHA_LOOKUP_GET_CACHE_VALUE_FAILURE + "_" + extractionNamespace.getNamespace(), 1));
+                    String cacheByteString = new String(lookupService.lookup(new LookupService.LookupData(extractionNamespace, key, valueColumn, decodeConfigOptional))).trim();
+                    if (secondaryColOverrideMap != null && secondaryColOverrideMap.containsKey(cacheByteString)) {
+                        return Strings.emptyToNull(secondaryColOverrideMap.get(cacheByteString));
+                    }
```

Comment:
In this case :  A->A_MAPPED, B->B_MAPPED.
LookupService will return the return A_MAPPED.

String cacheByteString = new String(lookupService.lookup(new LookupService.LookupData(extractionNamespace, key, valueColumn, decodeConfigOptional))).trim();
                    if (secondaryColOverrideMap != null && secondaryColOverrideMap.containsKey(cacheByteString)) {
                        return Strings.emptyToNull(secondaryColOverrideMap.get(cacheByteString));
                    }

thus we do not need above code, just return the value that you got from lookup service.